stl/src/StlLCMapStringW.cpp

```diff
@@ -39,10 +39,10 @@ extern "C" int __cdecl __crtLCMapStringW(_In_opt_z_ LPCWSTR const locale_name, _
         const int source_length = static_cast<int>(wcsnlen(source, source_count));
 
         // Include the null terminator if the source string is terminated within the buffer.
-        if (source_length < source_count) {
-            source_count = source_length + 1;
-        } else {
+        if (source_length > source_count) {
             source_count = source_length;
+        } else {
+            source_count = source_length + 1;

```

Comment:
This seems strange. When `source_length` happens to be equal to `source_count`, `source_count` would be incorrectly increased, which is possibly the reason of failures of tests.